### PR TITLE
fix(Button): render `labeled` as a <button>

### PR DESCRIPTION
**Pull request by x-shd**
Fix switch button to div 
#596

### CODE REVIEW

**Comment by levithomason** [comment]
Awesome, just need a test for this.  See `test/specs/elements/Button`.  Should be a very simply copy/paste update of one of the existing tests.

**Review by levithomason — CHANGES_REQUESTED** [review]
Awesome, just need a test for this. See `test/specs/elements/Button`. Should be a very simply copy/paste update of one of the existing tests.
Add your review

**Comment by x-shd** [comment]
Should work in production this trick fixed the problem.

File: test/specs/elements/Button/Button-test.js [hunk 1]
```diff
@@ -126,4 +126,13 @@ describe('Button', () => {
...
     })
   })
+
+  describe('labeled', () => {
+    it('is before the button and icon="plus" when labeled="left"', () => {
+      const wrapper = shallow(<Button labeled='left' icon='plus' />)
+      wrapper.should.have.tagName('button')
+      wrapper.should.have.className('labeled')
+      wrapper.should.have.exactly(1).descendants('Icon')
+    })
```
Comment (levithomason) [inline test/specs/elements/Button/Button-test.js:134]: Thanks for the test, let's simplify it a bit.  It is best to write tests that are as minimal as possible and assert one thing.  This makes them stronger and more maintainable.

In this case, most of the assertions made here are covered by other existing tests, all we should need to assert is that a `labeled` button renders a `button` element:

``` js
it('renders as a button', () => {
  shallow(<Button labeled />)
    .should.have.tagName('button')
})
```

**Review by levithomason — CHANGES_REQUESTED** [review]

File: test/specs/elements/Button/Button-test.js (hunk 1 above)
Comment (x-shd) [reply test/specs/elements/Button/Button-test.js:134]: Made

**Review by levithomason — APPROVED** [review]

**Comment by levithomason** [comment]
One linter issue:

image

**Comment by levithomason** [comment]
Also, if the description of this PR were updated slightly, it will auto close the related issue on merge:

```
Fix switch button to div 
#596
```

```
Fixes #596 switch button to div 
```

https://github.com/blog/1506-closing-issues-via-pull-requests

**Pull request merged**